joinmarket-webui/jam
# build(deps): upgrade to react 18

Finally updates to react 18.
This is actually the third attempt to do so - but there were always to many PRs waiting, which would have introduced many merge conflicts. Now in calmer times, it was picked up again.

Run `npm install` after checkout.

This updates most dependencies to their most recent versions.
Some behaviour had to be adapted (e.g. "success" step after importing a wallet), but all-in-all, it would work exactly as before.

Also noteworthy:
- had to insert a "success" step after wallet import flow as a workaround (`navigate` did not work anymore)
- finally "typescript-only" - disallow new js components

## :camera_flash: 

<img src="https://github.com/joinmarket-webui/jam/assets/3358649/4818ce7c-7742-431e-a1f8-20093ac6a4f4" width=450 />

## How to test
- Checkout the branch
- Run `npm install`
- Run `npm run dev:start`
- Verify everything is working as before (please test "create wallet", "import wallet", "sending", "fidelity bonds", etc.)
